### PR TITLE
Fixes #26129 - Add rebuild-config subcommand for hostgroups

### PR DESCRIPTION
**Pull request by treydock**

### CODE REVIEW

**Comment by treydock** [comment]
Made change ad-hoc to Satellite 6.4 test system for testing:

```
[root@satellite-test ~]# hammer hostgroup -h
Usage:
    hammer hostgroup [OPTIONS] SUBCOMMAND [ARG] ...

Parameters:
 SUBCOMMAND                    Subcommand
 [ARG] ...                     Subcommand arguments

Subcommands:
 ansible-roles                  
 create                        Create a host group
 delete                        Delete a host group
 delete-parameter              Delete parameter for a hostgroup
 info                          Show a host group
 list                          List all host groups
 puppet-classes                List all Puppet classes
 rebuild-config                Rebuild orchestration related configurations for hostgroup
 sc-params                     List all smart class parameters
 set-parameter                 Create or update parameter for a hostgroup
 smart-variables               List all smart variables
 update                        Update a host group

Options:
 -h, --help                    Print help
[root@satellite-test ~]# hammer hostgroup rebuild-config -h
Usage:
    hammer hostgroup rebuild-config [OPTIONS]

Options:
 --children-hosts CHILDREN_HOSTS         Operate on child hostgroup hosts
                                         One of true/false, yes/no, 1/0
 --id ID                                  
 --location LOCATION_NAME                Location name
 --location-id LOCATION_ID                
 --location-title LOCATION_TITLE         Location title
 --name NAME                             Hostgroup name
 --only ONLY                             Limit rebuild steps, valid steps are DHCP, DNS, TFTP
                                         Comma separated list of values. Values containing comma should be quoted or escaped with backslash
 --organization ORGANIZATION_NAME        Organization name
 --organization-id ORGANIZATION_ID       Organization ID
 --organization-title ORGANIZATION_TITLE Organization title
 --title TITLE                           Hostgroup title
 -h, --help                              Print help
[root@satellite-test ~]# hammer hostgroup rebuild-config --title 'base/owens/login' --only TFTP --children-hosts true
Configuration successfully rebuilt.

```

File: lib/hammer_cli_foreman/hostgroup.rb [hunk 1]
```diff
@@ -190,6 +190,15 @@ def validate_options
       end
     end
 
+    class RebuildConfigCommand < HammerCLIForeman::SingleResourceCommand
+      action :rebuild_config
+      command_name "rebuild-config"
+      desc _('Rebuild orchestration related configurations for hostgroup')
```
Comment (mbacovsky) [inline lib/hammer_cli_foreman/hostgroup.rb]: We prefer to keep as much descriptions from API docs as possible. Is this description providing any extra value over the original one ('Rebuild orchestration config')?

File: lib/hammer_cli_foreman/hostgroup.rb [hunk 2]
```diff
@@ -190,6 +190,15 @@ def validate_options
...
+      success_message _('Configuration successfully rebuilt.')
+
```
Comment (mbacovsky) [inline lib/hammer_cli_foreman/hostgroup.rb]: We usually require also failure_message. Is it missing on purpose?

File: lib/hammer_cli_foreman/hostgroup.rb (hunk 1 above)
Comment (treydock) [reply lib/hammer_cli_foreman/hostgroup.rb]: Probably not, honestly I just copied the code from `host.rb`.

File: lib/hammer_cli_foreman/hostgroup.rb (hunk 2 above)
Comment (treydock) [reply lib/hammer_cli_foreman/hostgroup.rb]: Not on purpose, just how this particular subcommand is defined for hosts which is where I copied the code from.

File: lib/hammer_cli_foreman/hostgroup.rb (hunk 1 above)
Comment (treydock) [reply lib/hammer_cli_foreman/hostgroup.rb]: Removed `desc`.

File: lib/hammer_cli_foreman/hostgroup.rb (hunk 2 above)
Comment (treydock) [reply lib/hammer_cli_foreman/hostgroup.rb]: Added `failure_message`

**Comment by treydock** [comment]
@mbacovsky Requested changes made

**Pull request merged**